aleph/search/parser.py

```diff
@@ -132,10 +132,12 @@ def __init__(self, args, authz, limit=None):
         # index.
         self.facet_names = set(self.getlist("facet"))
 
+        # Query to use for highlighting, defaults to the search query
+        self.highlight_text = self.get("highlight_text", self.text)
         # Include highlighted fragments of matching text in the result.
         self.highlight = self.getbool("highlight", False)
         self.highlight = self.highlight and settings.RESULT_HIGHLIGHT
-        self.highlight = self.highlight and self.text
+        self.highlight = self.highlight and self.highlight_text
```

Comment:
So we basically want to be able to highlight terms even when a query is not defined. I think I didn't fully understand that during our discussion the other day.
This looks great @tillprochaska!